TKGame/TKGame.cs

```diff
@@ -118,17 +116,34 @@ protected override void Update(GameTime gameTime)
             }
 
             // Exit the game if Escape is pressed
-            if (Keyboard.GetState().IsKeyDown(Keys.Escape))
+            if (Input.WasKeyPressed(Keys.Escape))
+            {
+                LevelEditor.SaveStageDataToJSON(currentStage, "auto_saved_stage_data");
                 Exit();
+            }
 
 #if DEBUG
             // Toggle the debug UI's visibility once per key press
             // TODO: Probably move this somewhere else
-            if (currentState.IsKeyDown(Keys.G) && !previousState.IsKeyDown(Keys.G))
```

Comment:
Don't see any kind of implementation to stop the game when Level Editor is open, may want to implement some way to stop gravity and stop enemy characters when Aaron merges the enemy class.

Reply:
I can add probably a single line to also pause the game when G is pressed, that's not a problem.